syft/cataloger/java/archive_parser.go

```diff
@@ -156,6 +160,7 @@ func (j *archiveParser) discoverMainPackage() (*pkg.Package, error) {
 
 // discoverPkgsFromPomProperties parses Maven POM properties for a given parent package, returning all listed Java packages found and
 // associating each discovered package to the given parent package.
+// nolint:funlen,gocognit
```

Comment:
This isn't a request for change, but a note for discussion at some point in the future — I think we might be getting into that stage of the code base's life where functions are slowly entering the large/complex territory. I'd love to see us discuss this as a group at some point and think about ways to extract functions and keep the code as readily understandable as possible

Reply:
My default stance when I run into these points is to break it up into smaller helper functions (which I've had to a few times now)... this one seemed rather non-obvious in how to do so, so punted until the new pattern is more obvious. This time I added more comments to help compensate.

Reply:
The comments are great. Definitely better than not having them. If you want to talk through this function, just as an exercise, let me know. But to me, it's not something that should block this PR